planetiler-core/src/main/java/com/onthegomap/planetiler/Planetiler.java

```diff
@@ -358,8 +361,18 @@ public Planetiler addShapefileSource(String projection, String name, Path defaul
   public Planetiler addGeoPackageSource(String name, Path defaultPath, String defaultUrl) {
     Path path = getPath(name, "geopackage", defaultPath, defaultUrl);
     return addStage(name, "Process features in " + path,
-      ifSourceUsed(name,
-        () -> GeoPackageReader.process(name, List.of(path), featureGroup, config, profile, stats)));
+      ifSourceUsed(name, () -> {
+        List<Path> sourcePaths = List.of(path);
+        if (FileUtils.hasExtension(path, "zip")) {
+          sourcePaths = FileUtils.unzipMatchingFiles(path, "*.gpkg", tmpDir);
+        }
```

Comment:
Can we push the unzip handling inside geopackage reader so addGeoPackageSource enumerates the files in the zip file and passes that list of paths into the geopackage reader, then geopackage reader extracts them first if necessary?

That would move toward making the reader usage more consistent and encapsulate the logic of whether they can read into a zip file or need to extract first inside the reader.  It would also let the reader explicitly delete the unzipped file on close to free up disk space as soon as it's not needed since we can't always trust deleteOnExit to happen.